### ISSUE TITLE
Failing to deploy to Pypi on Travis

### ISSUE THREAD

**Issue by gilesknap** [dls-controls/versiongit#1]
Hey Tom,

Have you seen this one? Pypi deploy is failing with:-
```
copying gphotos_sync.egg-info/entry_points.txt -> gphotos-sync-2.12.2/gphotos_sync.egg-info
copying gphotos_sync.egg-info/requires.txt -> gphotos-sync-2.12.2/gphotos_sync.egg-info
copying gphotos_sync.egg-info/top_level.txt -> gphotos-sync-2.12.2/gphotos_sync.egg-info
Writing gphotos-sync-2.12.2/setup.cfg
error: [Errno 2] No such file or directory: 'gphotos-sync-2.12.2/test/_version_static.py'
InvalidDistribution: Cannot find file (or expand pattern): 'dist/*'
gphotos/__pycache__/__init__.cpython-37.pyc already exists, no checkout
gphotos/__pycache__/_version_git.cpython-37.pyc already exists, no checkout
error: could not restore untracked files from stash
The stash entry is kept in case you need it again.
PyPI upload failed.
```
But setup.py install and pipenv install work fine. I do see `test/_version_static.py` in my build folder after running setup.py locally.

My setup.py, __init__.py etc. is here https://github.com/gilesknap/gphotos-sync/tree/versioning

The test folder is supposed to be excluded from the distribution so perhaps I have done something incorrect there.

**Comment by thomascobb** [comment]
I've commented in https://github.com/gilesknap/gphotos-sync/pull/193

**Comment by gilesknap** [comment]
Thanks.
That did not appear to fix the issue. I'll copy pymalcolm approach to not deploying the test package and see if that helps.

**Comment by gilesknap** [comment]
- I have removed test from the deployed packages.
- I cloned versiongit and run version-git command to verify my install.
  - I noticed that .gitattributes was missing
  - everything else looks OK 

Now the deploy to Pypi does work but its _version_static.py contains `0+unknown.error`

version-git works fine everywhere except when building on travis for pypi.

@thomascobb 
Any ideas?

**Comment by gilesknap** [comment]
This local install looks ok:
```
(versioning) [giles@gklinux gphotos-sync]$ pipenv --rm
Removing virtualenv (/home/giles/.local/share/virtualenvs/gphotos-sync-23K6u7j6)…
(versioning) [giles@gklinux gphotos-sync]$ 
(versioning) [giles@gklinux gphotos-sync]$ pipenv install .
Creating a virtualenv for this project…
...
Installing .…
⠸ Installing...installing version 2.12.7
Adding gphotos-sync to Pipfile's [packages]…
✔ Installation Succeeded 
Installing dependencies from Pipfile.lock (2ab7d9)…
Ignoring pywin32: markers 'platform_system == "Windows"' don't match your environment
Looking in indexes: https://pypi.python.org/simple
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 15/15 — 00:00:02
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
(versioning) [giles@gklinux gphotos-sync]$ find /home/giles/.local/share/virtualenvs/gphotos-sync-23K6u7j6 -name _version_static.py -exec cat {} \;
__version__ = '2.12.7'
(versioning) [giles@gklinux gphotos-sync]$ 
```

**Comment by gilesknap** [comment]
This was fixed in #2

**Comment by thomascobb** [comment]
I've released versiongit 0.5 to pypi that includes your fix, but it also drops the usage of `_version_static.py` and modifies `_version_git.py` inplace instead, so you will have to modify your `__init__.py` to use it.

You can add it by doing `pip install versiongit; versiongit /path/to/repo/modulename`

**Comment by gilesknap** [comment]
updated and working on gphotos-sync.
